MAUI/DataGrid/row-drag-and-drop.md

```diff
@@ -0,0 +1,360 @@
+
+private void DataGrid_QueryRowDragging(object? sender, Syncfusion.Maui.DataGrid.DataGridQueryRowDraggingEventArgs e)
+{
+    var totalHeight = dataGrid.RowColumnIndexToPoint(new RowColumnIndex(viewModel.Orders.Count, 0)).Y + this.dataGrid.RowHeight;
+    if (e.DraggingAction == DataGridDragAction.DragEnded)
+    {
+        if (Math.Ceiling(e.Position.Y + (dataGrid.RowHeight / 2)) > totalHeight && e.To == viewModel.Orders.Count)
+        {
+            //Will hit if the row is dropped at the last position                 
+            DisplayAlert("RowDragAndDrop info", "The row is dropped at the last position", "OK");
+        }
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+
+## Updating summaries when dragging and dropping a row between groups
+
+Grouping and summarization of items in the data grid are determined by the group key. When an item is dragged from one group to another, the group key of the dragged item will differ from the group key of the items in the dropped group. Therefore, by default, the summaries will not be updated. This reflects the actual behavior of this control.
+
+Therefore, to update the summaries when a row is dragged and dropped between groups, call the `UpdateCaptionSummaries` and `Refresh` methods in the `QueryRowDragging` event.
+
+{% tabs %}
+{% highlight c#%}
+public partial class MainPage : ContentPage
+{
+    private SfDataGrid dataGrid;
+    private ViewModel viewModel;
+    public MainPage()
+    {
+         InitializeComponent();
+         dataGrid = new SfDataGrid();
+         viewModel = new OrderRepo();
+         dataGrid.ItemsSource = viewModel.Orders;
+         dataGrid.QueryRowDragging += DataGrid_QueryRowDragging;
+         this.Content = dataGrid;
+    }
+   private void DataGrid_QueryRowDragging(object? sender, Syncfusion.Maui.DataGrid.DataGridQueryRowDraggingEventArgs e)
+    {
+         if (e.DraggingAction == DataGridDragAction.DragEnded)
+         {
+             // Delay is given for refreshing the view.
+             await Task.Delay(100);                
+             this.dataGrid.View.TopLevelGroup.UpdateCaptionSummaries();
+             this.dataGrid.View.Refresh();
+         }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+
+## Cancel auto scrolling 
+
+The vertical auto-scrolling of the `SfDataGrid` during row drag-and-drop can be canceled using the `CanAutoScroll` argument in the `QueryRowDragging` event handler.
+
+{% tabs %}
+{% highlight c# %}
+
+dataGrid.QueryRowDragging += DataGrid_QueryRowDragging;
+
+private async void DataGrid_QueryRowDragging(object? sender, Syncfusion.Maui.DataGrid.DataGridQueryRowDraggingEventArgs e)
+{
+   e.CanAutoScroll = true;
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+
+## Customization of draw view
+
+The row view’s background color, text color, and indicator color can be customized using the `RowDragViewBackgroundColor`, `RowDragViewTextColor`, and `RowDraggingIndicatorLineColor` properties.
```

Comment:
The row drag view’s background color, text color, and indicator color can be customized using the `RowDragViewBackgroundColor`, `RowDragViewTextColor`, and `RowDraggingIndicatorLineColor` properties.